msbuild/dbuild/CompileD.cs

```diff
@@ -66,31 +106,508 @@ public virtual ITaskItem[] Sources
             }
         }
 
-        public string CommandLineTemplate
+        // Hidden
+        public bool DoNotLink
         {
-            get;
-            set;
+            get { return GetBoolProperty("DoNotLink"); }
+            set
+            {
+                SetBoolProperty("DoNotLink", "Do Not Link",
+                                "Compile only. Do not link (-c)",
+                                "-c", value);
+            }
         }
 
-        public bool ShowCommandLine
+        public string CodeGeneration
         {
-            get;
-            set;
+            get { return GetStringProperty("CodeGeneration"); }
+            set
+            {
+                string[][] switchMap = new string[3][]
+                {
+                    new string[2] { "32BitsMS-COFF", "-m32mscoff" },
+                    new string[2] { "32Bits", "-m32" },
+                    new string[2] { "64Bits", "-m64" }
+                };
+
+                SetEnumProperty("CodeGeneration", "Code Generation",
+                                "Generate 32 or 64 bit code.",
+                                switchMap, value);
+            }
         }
 
-        public string Xaml
+        // General
+        public string[] ImportPaths
         {
-            get;
-            set;
+            get { return GetStringArray("ImportPaths"); }
+            set
+            {
+                SetStringArray("ImportPaths", ToolSwitchType.StringPathArray, "Import Paths",

```

Comment:
StringPathArray is not available in V2013, so please use StringArray for that version.